From 713f0d3793452f4fccf30e32847b20e2c6ba2cf1 Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Wed, 25 Apr 2018 17:06:33 -0400
Subject: [PATCH 5658/5725] drm/amdkfd: Remove IH patching workaround for
 Vega10

Early CP firmware during bring-up failed to set the pasid in the IH
ring entries. We had a racy driver workaround at the time. Current
production firmware no longer requires this hack.

Change-Id: Iccd0a4412918645e0b985be9eb9bb2aaeb486d37
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_int_process_v9.c | 49 ++++---------------------
 1 file changed, 7 insertions(+), 42 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_int_process_v9.c b/drivers/gpu/drm/amd/amdkfd/kfd_int_process_v9.c
index 009d6f4..728aaad 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_int_process_v9.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_int_process_v9.c
@@ -25,24 +25,12 @@
 #include "soc15_int.h"
 
 
-static uint32_t kfd_get_pasid_from_vmid(struct kfd_dev *dev, uint8_t vmid)
-{
-	uint32_t pasid = 0;
-	const struct kfd2kgd_calls *f2g = dev->kfd2kgd;
-
-	if (f2g->get_atc_vmid_pasid_mapping_valid(dev->kgd, vmid))
-		pasid = f2g->get_atc_vmid_pasid_mapping_pasid(dev->kgd, vmid);
-
-	return pasid;
-}
-
 static bool event_interrupt_isr_v9(struct kfd_dev *dev,
 					const uint32_t *ih_ring_entry,
 					uint32_t *patched_ihre,
 					bool *patched_flag)
 {
 	uint16_t source_id, client_id, pasid, vmid;
-	bool result = false;
 
 	source_id = SOC15_SOURCE_ID_FROM_IH_ENTRY(ih_ring_entry);
 	client_id = SOC15_CLIENT_ID_FROM_IH_ENTRY(ih_ring_entry);
@@ -59,36 +47,13 @@ static bool event_interrupt_isr_v9(struct kfd_dev *dev,
 			 data[4], data[5], data[6], data[7]);
 	}
 
-	if ((vmid >= dev->vm_info.first_vmid_kfd &&
-	     vmid <= dev->vm_info.last_vmid_kfd) &&
-	    (source_id == SOC15_INTSRC_CP_END_OF_PIPE ||
-	     source_id == SOC15_INTSRC_SDMA_TRAP ||
-	     source_id == SOC15_INTSRC_SQ_INTERRUPT_MSG ||
-	     source_id == SOC15_INTSRC_CP_BAD_OPCODE ||
-	     client_id == SOC15_IH_CLIENTID_VMC ||
-	     client_id == SOC15_IH_CLIENTID_UTCL2)) {
-
-		/*
-		 * KFD want to handle this INT, but MEC firmware did
-		 * not send pasid. Try to get it from vmid mapping
-		 * and patch the ih entry. It's a temp workaround.
-		 */
-		WARN_ONCE((!pasid), "Fix me.\n");
-		if (!pasid) {
-			uint32_t temp = le32_to_cpu(ih_ring_entry[3]);
-
-			pasid = kfd_get_pasid_from_vmid(dev, vmid);
-			memcpy(patched_ihre, ih_ring_entry,
-			       dev->device_info->ih_ring_entry_size);
-			patched_ihre[3] = cpu_to_le32(temp | pasid);
-			*patched_flag = true;
-		}
-		result = pasid ? true : false;
-	}
-
-	/* Do not process in ISR, just request it to be forwarded to WQ. */
-	return result;
-
+	return (pasid != 0) &&
+		(source_id == SOC15_INTSRC_CP_END_OF_PIPE ||
+		 source_id == SOC15_INTSRC_SDMA_TRAP ||
+		 source_id == SOC15_INTSRC_SQ_INTERRUPT_MSG ||
+		 source_id == SOC15_INTSRC_CP_BAD_OPCODE ||
+		 client_id == SOC15_IH_CLIENTID_VMC ||
+		 client_id == SOC15_IH_CLIENTID_UTCL2);
 }
 
 static void event_interrupt_wq_v9(struct kfd_dev *dev,
-- 
2.7.4

